File: backend/app/schemas/elder.py
from datetime import date, datetime
from pydantic import BaseModel, Field
from typing import Optional, List


class ElderBase(BaseModel):
    name: str
    gender: str
    birth_date: date
    id_card: str
    phone: Optional[str] = None
    emergency_contact: Optional[str] = None
    emergency_phone: Optional[str] = None
    address: Optional[str] = None
    health_status: str = "stable"
    care_level: str = "basic"
    room_number: Optional[str] = None
    bed_number: Optional[str] = None
    admission_date: Optional[date] = None
    medical_history: Optional[str] = None
    allergies: Optional[str] = None
    dietary_restrictions: Optional[str] = None
    mobility_level: Optional[str] = None
    cognitive_level: Optional[str] = None
    status: str = "active"
    avatar_url: Optional[str] = None
    remark: Optional[str] = None


class ElderCreate(ElderBase):
    pass


class ElderUpdate(BaseModel):
    name: Optional[str] = None
    gender: Optional[str] = None
    birth_date: Optional[date] = None
    phone: Optional[str] = None
    emergency_contact: Optional[str] = None
    emergency_phone: Optional[str] = None
    address: Optional[str] = None
    health_status: Optional[str] = None
    care_level: Optional[str] = None
    room_number: Optional[str] = None
    bed_number: Optional[str] = None
    admission_date: Optional[date] = None
    medical_history: Optional[str] = None
    allergies: Optional[str] = None
    dietary_restrictions: Optional[str] = None
    mobility_level: Optional[str] = None
    cognitive_level: Optional[str] = None
    status: Optional[str] = None
    avatar_url: Optional[str] = None
    remark: Optional[str] = None


class ElderResponse(ElderBase):
    id: int
    created_by: Optional[int] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class ElderStatusUpdate(BaseModel):
    status: str
    remark: Optional[str] = None

File: frontend/src/api/elder.ts
import api from './client'

export interface Elder {
  id: number
  name: string
  gender: string
  birth_date: string
  id_card: string
  phone?: string
  emergency_contact?: string
  emergency_phone?: string
  address?: string
  health_status: string
  care_level: string
  room_number?: string
  bed_number?: string
  admission_date?: string
  medical_history?: string
  allergies?: string
  dietary_restrictions?: string
  mobility_level?: string
  cognitive_level?: string
  status: string
  avatar_url?: string
  remark?: string
  created_by?: number
  created_at: string
  updated_at: string
}

export interface PaginatedResponse<T> {
  total: number
  page: number
  page_size: number
  items: T[]
}

export const elderApi = {
  getList: (params?: {
    page?: number
    page_size?: number
    keyword?: string
    status?: string
    health_status?: string
    care_level?: string
  }) => {
    return api.get<PaginatedResponse<Elder>>('/elders', { params })
  },

  getDetail: (id: number) => {
    return api.get<Elder>(`/elders/${id}`)
  },

  create: (data: Partial<Elder>) => {
    return api.post<Elder>('/elders', data)
  },

  update: (id: number, data: Partial<Elder>) => {
    return api.put<Elder>(`/elders/${id}`, data)
  },

  updateStatus: (id: number, status: string, remark?: string) => {
    return api.patch<Elder>(`/elders/${id}/status`, { status, remark })
  },

  remove: (id: number) => {
    return api.delete(`/elders/${id}`)
  },
}
